datasets/me/acp_peps/crawler.py

```diff
@@ -0,0 +1,162 @@
+import io
+from csv import DictReader
+from datetime import datetime
+from typing import List, Dict, Optional, Tuple
+from zipfile import ZipFile, BadZipFile
+
+from zavod import Context, helpers as h
+from zavod.logic.pep import categorise
+from zavod.shed.trans import (
+    apply_translit_full_name,
+    make_position_translation_prompt,
+)
+
+TRANSLIT_OUTPUT = {
+    "eng": ("Latin", "English"),
+}
+POSITION_PROMPT = prompt = make_position_translation_prompt("cnr")
+
+
+def extract_latest_filing(
+    dates: List[Dict[str, object]]
+) -> Tuple[Optional[str], Optional[int]]:
+    if not dates:
+        return None, None
+
+    # Sort the dates in descending order
+    dates_sorted = sorted(
+        dates,
+        key=lambda item: datetime.strptime(item["datum"], "%Y-%m-%d"),
+        reverse=True,
+    )
+
+    # Assume the first entry is the latest
+    latest_entry = dates_sorted[0]
+    latest_date: Optional[str] = latest_entry.get("datum")
+    latest_report_id: Optional[int] = latest_entry.get("stariIzvjestaj")
+
+    # Check if the latest report is -1, move to the next valid entry
+    if latest_report_id == -1:
+        for entry in dates_sorted[1:]:
+            if entry.get("stariIzvjestaj") != -1:
+                latest_date = entry.get("datum")
+                latest_report_id = entry.get("stariIzvjestaj")
+                break
+        else:
+            # If all entries have -1, set the value to None
+            latest_report_id = None
+
+    return latest_date, latest_report_id
+
+
+def fetch_and_process_csv(context: Context, latest_report_id: int):
+    # URL for fetching the CSV
+    url = f"https://portal.antikorupcija.me:9343/acamPublic/izvestajDetailsCSV.json?izvestajId={latest_report_id}"
+
+    # Fetch the ZIP file containing the CSV
+    zip_path = context.fetch_resource(f"{latest_report_id}.zip", url)
+    try:
+        with ZipFile(zip_path, "r") as zip:
+            # Find the file name that matches the pattern
+            file_name = next(
+                (
+                    name
+                    for name in zip.namelist()
+                    if name.startswith("csv_funkcije_funkcionera_")
+                ),
+                None,
+            )
```

Comment:
This is on the edge of being too clever. But It works and makes sense once I've processed what it does, and might make for tidy code once the expression is broken up a bit, so I think it's ok.

The nesting is a bit awkward to read so I'd suggest assigning the filtered list first and calling next on the next line or something?